### ISSUE TITLE
"Waiting for Images" - system hung busy waiting on bStepByStep in Tracking::Track

### ISSUE THREAD

**Issue by stellarpower** [UZ-SLAMLab/ORB_SLAM3#405]
Hi,

I'm running a modification of one of the examples to do tracking on simple monocular videos - modified mainly because our data is in a different format from the given datasets. This was working fine previously, but now is sporadically hanging constantly with the Pangolin window saying "Waiting for Images". 

Quite possibly there is an issue in my code, or, we have some bad frames in the input files, as the choice of input file seems to make a differnece. However, it's hard to debug without knowing what's really going on, and, being stuck busy waiting without an error, I'm wondering if this could have opened up something that could be improved.

Please could I ask a bit more as to what the step-by-step mode is doing, and where/what is meant to alter the mbStep so that it exits [the loop](https://github.com/UZ-SLAMLab/ORB_SLAM3/blob/578500b836ea05be40d6188b815bb0ddbcdc75f1/src/Tracking.cc#L1663)? I'll then get some debugging information, and ascertain as to if my code is at fault or if there's another issue undenreath too.

Thanks!
